app/controllers/api/v2/roles_controller.rb

```diff
@@ -39,7 +41,10 @@ def create
       param_group :role
 
       def update
-        process_response @role.update_attributes(role_params)
+        taxonomies = taxonomy_params!
+        result = @role.update_attributes(role_params)
+        @role.set_taxonomies(taxonomies)
```

Comment:
Yes, I do not permit these parameters in `role_params_filter` because I don't want them to be blindly passed to update, they have to be sanitized based on what current user's taxonomies are. It is sort of an authorization that we normally do in controller, so I take these parameters and let `set_taxonomies` handle it.

I don't want to let role save triggering also the taxonomies and filters change, note that the result is different for each user. Also it requires `User.current` to be set. This indicates that model itself can't handle it without user session. I know we have current user stored in thread so we can access it in the model layer and I leverage this hack but it does not mean that it's a good idea. Normally we'd have to pass user instance to `set_taxonomies` method so it would have to be called from controller explicitly.

If you want, I can wrap that into single model method so the caller can decide if he wants to run only `update_attributes(attrs)` or `update_attributes_and_set_taxonomies(attrs, tax_attrs)`.